src/v/config/constraint_types.cc

```diff
@@ -0,0 +1,209 @@
+  , _methods{std::move(methods)} {}
+
+const ss::sstring& constraint::name() const { return _args.name; }
+const constraint_type& constraint::type() const { return _args.type; }
+bool constraint::is_supported() const { return _methods.is_supported; }
+
+bool constraint::validate(const cluster::topic_configuration& topic_cfg) const {
+    return _methods.validator(topic_cfg, _args);
+}
+
+void constraint::clamp(cluster::topic_configuration& topic_cfg) const {
+    return _methods.clamper(topic_cfg, _args);
+}
+} // namespace config
+
+namespace YAML {
+
+template<typename T>
+void encode_range(Node& node, config::range_values<T>& range) {
+    if (range.min) {
+        node["min"] = *range.min;
+    }
+    if (range.max) {
+        node["max"] = *range.max;
+    }
+}
+
+Node convert<config::constraint_args>::encode(const type& rhs) {
+    Node node;
+    node["name"] = rhs.name;
+    node["type"] = ss::sstring(config::to_string_view(rhs.type));
+
+    ss::visit(
+      rhs.flags,
+      [&node](config::range_values<int64_t> signed_range) {
+          encode_range(node, signed_range);
+      },
+      [&node](config::range_values<uint64_t> unsigned_range) {
+          encode_range(node, unsigned_range);
+      },
+      [&node](config::constraint_enabled_t enabled) {
+          node["enabled"] = enabled == config::constraint_enabled_t::yes;
+      });
+
+    return node;
+}
+
+template<typename T>
+config::range_values<T> decode_range(const Node& node) {
+    config::range_values<T> range;
+    if (node["min"] && !node["min"].IsNull()) {
+        range.min = node["min"].as<T>();
+    }
+
+    if (node["max"] && !node["max"].IsNull()) {
+        range.max = node["max"].as<T>();
+    }
+    return range;
+}
+
+template<typename T>
+bool maybe_set_range(const Node& node, config::constraint_args& args) {
+    auto range = decode_range<T>(node);
+    if (!range.min && !range.max) {
+        return false;
+    }
+
+    args.flags = range;
+    return true;
+}
+
+bool convert<config::constraint_args>::decode(const Node& node, type& rhs) {
+    for (const auto& s : {"name", "type"}) {
+        if (!node[s]) {
+            return false;
+        }
+    }
+
+    auto name = node["name"].as<ss::sstring>();
+    auto type = config::from_string_view<config::constraint_type>(
+                  node["type"].as<ss::sstring>())
+                  .value();
+
+    rhs.name = std::move(name);
+    rhs.type = std::move(type);
+
+    return config::ternary_property_op(
+      config::shard_local_cfg().get(rhs.name),
+      [&node, &rhs] { return maybe_set_range<int64_t>(node, rhs); },
+      [&node, &rhs] { return maybe_set_range<uint64_t>(node, rhs); },
+      [&node, &rhs] {
+          // For "enabled" contraints
+          if (node["enabled"] && !node["enabled"].IsNull()) {
+              rhs.flags = config::constraint_enabled_t(
+                node["enabled"].as<bool>());
+              return true;
+          }
+
+          return false;
```

Comment:
In your scenario, enabled is read as Null by the YAML parser and parsing would fail and a message is automatically logged. In other comments I provided examples of what the error looks like. I also included sections for YAML parsing failures in the RFC.